### PR TITLE
split language agnostic to a separate file

### PR DESCRIPTION
**Pull request by eshellman**
## What does this PR do?
It moves the "Language Agnostic" books to a separate file.
Implements #5249 

## Checklist:
- [ x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [ x] Put lists in alphabetical order, correct spacing.
- [x ] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by eshellman** [comment]
hmmm seems to break the link checker.
we should settle on a crosslink syntax so the obvious cross-links can be made.

**Comment by eshellman** [comment]
as this will be a massive conflict creator, we should decide whether to procede with it before friday.

**Comment by davorpa** [comment]
> hmmm seems to break the link checker.
> we should settle on a crosslink syntax so the obvious cross-links can be made.

This is a known issue the first time when a new branch is created. It's produced by the github action in charge of find what files was change and then, scope the links to search with the bot.

https://github.com/EbookFoundation/free-programming-books/blob/b76cdd789a68b30062b6628cadb80f9c0c7de41a/.github/workflows/main.yml#L6-L12

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -0,0 +1,761 @@
...
+* [Red Gate Books](http://www.red-gate.com/community/books/index)
+* [Stef's Free Online Smalltalk Books](http://stephane.ducasse.free.fr/FreeBooks/)
+* [TechBeamers.com](http://www.techbeamers.com)
+* [TechBooksForFree.com](http://www.techbooksforfree.com)
+* [The Definitive C++ Book Guide and List](http://stackoverflow.com/questions/388242/the-definitive-c-book-guide-and-list)
+* [Theassayer.org](http://theassayer.org)
+* [tutorialspoint](http://www.tutorialspoint.com)
+
+
+#### Algorithms & Data Structures
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:83]: After moving, **heading dept should be decremented by one `###`** to be at the same level that metalist section and then, mantain the TOC synchronized :rocket:

**Apply on all headers with `####`** :wink:

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -0,0 +1,761 @@
...
+* [Learning Modern 3D Graphics Programming](https://web.archive.org/web/20150225192611/http://www.arcsynthesis.org/gltut/index.html) - Jason L. McKesson (draft)
+* [Notes for a Computer Graphics Programming Course](https://www.cs.csustan.edu/~rsc/CS3600F00/Notes.pdf) - Dr. Steve Cunningham (PDF)
+* [OpenGL](http://www.songho.ca/opengl/index.html) - Concepts and illustrations.
+* [Ray Tracing in One Weekend](https://raytracing.github.io) - Peter Shirley (HTML)
+* [ShaderX series](http://www.realtimerendering.com/resources/shaderx/) - Wolfgang Engel
+* [Tutorials for modern OpenGL](http://www.opengl-tutorial.org)
+* [WebGL Insights](http://webglinsights.com) - Patrick Cozzi and Contributors
+
+
+### Graphical User Interfaces
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:258]: Nice! Another with level 3.

File: books/free-programming-books-subjects.md [hunk 3]
```diff
@@ -0,0 +1,761 @@
...
+* [2D Game Development: From Zero To Hero](https://github.com/Penaz91/2DGD_F0TH) - Daniele Penazzo (HTML, [PDF, EBPUB, Kindle...](https://therealpenaz91.itch.io/2dgd-f0th#download)) (:construction: *in process*)
+* [Coding With Minecraft](https://turtleappstore.com/book/) - Al Sweigart
+* [Designing Virtual Worlds](https://mud.co.uk/richard/DesigningVirtualWorlds.pdf) - Richard A. Bartle (PDF)
+* [Game AI Pro](http://www.gameaipro.com) - Steve Rabin
+* [Game Design with AGS](https://ensadi.github.io/AGSBook/) - Dave Ensminger, A. G. Madi
+* [Game Programming Patterns](http://gameprogrammingpatterns.com) - Bob Nystrom
+* [Procedural Content Generation in Games](http://pcgbook.com) - Noor Shaker, Julian Togelius, Mark Nelson
+
+
+### Graphics Programming
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:235]: It's OK using level 3.

File: books/free-programming-books-subjects.md [hunk 4]
```diff
@@ -0,0 +1,761 @@
...
+* [Yet Another Introductory Number Theory Textbook](https://www.poritz.net/jonathan/share/yaintt.pdf) - Jonathan A. Poritz (PDF)
+
+
+#### Mathematics For Computer Science
+
+* [A Mathematical Theory of Communication](https://archive.org/details/bstj27-4-623) - Claude E.Shannon
+* [Discrete Structures for Computer Science: Counting, Recursion, and Probability](http://cglab.ca/~michiel/DiscreteStructures/) - Michiel Smid
+
+
+#### Misc
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:446]: Maybe move entire section to the top, below `0 - Metalists` :question:

File: books/free-programming-books-subjects.md [hunk 5]
```diff
@@ -0,0 +1,761 @@
...
+* [Real-World Maintainable Software](https://www.oreilly.com/ideas/real-world-maintainable-software) - Abraham Marin-Perez
+* [Record-Playback Test Automation: Sahi & Selenium IDE: Critical Evaluation of Record-Playback Automation Tools](https://leanpub.com/manualToAutomatedWithSeleniumIDEAndSahi) - Shashikant Jagtap  *(Leanpub account or valid email requested)*
+* [Scientific Programming and Computer Architecture](https://divakarvi.github.io/bk-spca/spca.html) - Divakar Viswanath
+* [Signal Computing: Digital Signals in the Software Domain](http://faculty.washington.edu/stiber/pubs/Signal-Computing/Signal%20Computing.pdf) - Michael Stiber, Bilin Zhang Stiber, Eric C. Larson (PDF)
+* [Small Memory Software](http://www.smallmemory.com/book.html)
+* [Web Almanac](https://almanac.httparchive.org/static/pdfs/web_almanac_2019_en.pdf) (PDF)
+* [Writing Native Mobile Apps in a Functional Language Succinctly](https://www.syncfusion.com/ebooks/writing_native_mobile_apps_in_a_functional_language_succinctly) - Vassili Kaplan
+
+
+#### MOOC
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:513]: Maybe move entire section to the top, below `0 - Metalists` :question:

Its more like platforms than subject

File: books/free-programming-books-subjects.md [hunk 6]
```diff
@@ -0,0 +1,761 @@
...
+* [The 30 Minute Regex Tutorial](http://www.codeproject.com/Articles/9099/The-Minute-Regex-Tutorial) - Jim Hollenhorst
+* [The Bastards Book of Regular Expressions: Finding Patterns in Everyday Text](https://leanpub.com/bastards-regexes) - Dan Nguyen  *(Leanpub account or valid email requested)*
+
+
+#### Reverse Engineering
+
+* [BIOS Disassembly Ninjutsu Uncovered 1st Edition](http://bioshacking.blogspot.co.uk/2012/02/bios-disassembly-ninjutsu-uncovered-1st.html) - Darmawan Salihun (PDF)
+* [Hacking the Xbox: An Introduction to Reverse Engineering](https://www.nostarch.com/xboxfree/) - Andrew "bunnie" Huang
+* [iOS App Reverse Engineering](https://github.com/iosre/iOSAppReverseEngineering) (PDF)
+* [Reverse Engineering for Beginners](http://beginners.re) - Dennis Yurichev (PDF)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:647]: Remember remove it. See:
- #5558 

I'm working on add those localized versions that are free

File: books/free-programming-books-subjects.md [hunk 7]
```diff
@@ -0,0 +1,761 @@
...
+#### Web Performance
+
+* [Book of Speed](http://www.bookofspeed.com) - Stoyan Stefanov
+* [Designing for Performance](http://designingforperformance.com) - Lara Hogan
+* [High Performance Accelerated Websites](https://thisisyuu.github.io/ebook) - Anshul (HTML) (:construction: *in process*)
+* [High Performance Browser Networking](https://hpbn.co) - Ilya Grigorik
+* [Mature Optimization](http://carlos.bueno.org/optimization/mature-optimization.pdf) - Carlos Bueno (PDF)
+
+
+### Web Services
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:753]: It's OK. Other with level 3.

File: books/free-programming-books-subjects.md [hunk 8]
```diff
@@ -0,0 +1,761 @@
...
+
+* [RESTful Web Services](http://restfulwebapis.org/RESTful_Web_Services.pdf) (PDF)
+
+
+### Workflow
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:758]: It's right. Level 3.

**Review by davorpa — CHANGES_REQUESTED** [review]
All `####` headings should be fixed to `###` with the propose of mantaining TOC synchronized. :hugs:

/labelthis: breaking changes

**Review by davorpa — APPROVED** [review]

**Pull request merged**